src/programflow-visualization/types.ts

```diff
@@ -7,6 +7,7 @@ type Failure = { errorMessage: string };
 
 // State Types for the Frontend
 type FrontendTrace = Array<FrontendTraceElem>;
+// FIXME: what are this array elements? Why isn't there a type with named fields?
```

Comment:
See `generateHTML` in HTMLGenerator.ts: it corresponds to a single rendered visualization step. It has the following elements:
- Line number
- the stack rendered as an HTML fragment
- the heap rendered as an HTML fragment
- current filename
- the current state of the output

I don't know why there isn't a type with named fields. It was already solved this way before I integrated the visualization into the WYPP plugin and to be honest, I just did not fix it because it already worked and my main main focus was to get everything working.